docs/admin/federation/index.md

```diff
@@ -0,0 +1,184 @@
+
+```shell
+$ kubectl get deployments --namespace=federation
+NAME                            DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+federation-apiserver            1         1         1            1           1m
+federation-controller-manager   1         1         1            1           1m
+```
+
+Running `federation-up.sh` also creates a new record in our kubeconfig for us
+to be able to talk to federation apiserver. We can view this by running
+`kubectl config view`.
+
+## Registering Kubernetes clusters for federation
+
+Now that we have the federation control plane up and running, we can start registering Kubernetes clusters.
+
+First of all, we need to create a secret containing kubeconfig for that Kubernetes cluster, which federation control plane will use to talk to that Kubernetes cluster.
+For now, we create this secret in the host Kubernetes cluster (that hosts federation control plane). When we start supporting secrets in federation control plane, we will create this secret there.
+Suppose that our kubeconfig for Kubernetes cluster is at `/cluster1/kubeconfig`, we can run the following command to create the secret:
+
+```shell
+$ kubectl create secret generic cluster1 --namespace=federation --from-file=/cluster1/kubeconfig
+```
+
+Note that the file name should be `kubeconfig` since file name determines the name of the key in the secret.
+
+Now that the secret is created, we are ready to register the cluster. The YAML file for cluster will look like:
+
+```yaml
+apiVersion: v1beta1
+kind: Cluster
+metadata:
+  name: cluster1
+spec:
+  serverAddressByClientCIDRs:
+  - clientCIDR: <client-cidr>
+    serverAddress: <apiserver-address>
+  secretRef:
+    name: <secret-name>
+```
+
+We need to insert the appropriate values for `<client-cidr>`, `<apiserver-address>` and `<secret-name>`.
+`<secret-name>` here is name of the secret that we just created.
+serverAddressByClientCIDRs contains the various server addresses that clients
+can use as per their CIDR. We can set the server's public IP address with CIDR
+`"0.0.0.0/0"` which all clients will match. In addition, if we want internal
+clients to use server's clusterIP, we can set that as serverAddress. The client
+CIDR in that case will be a CIDR that only matches IPs of pods running in that
+cluster.
+
+Assuming our YAML file is located at `/cluster1/cluster.yaml`, we can run the following command to register this cluster:
+
+```shell
+$ kubectl create -f /cluster1/cluster.yaml --cluster=federation-cluster
+
+```
+
+By specifying `--cluster=federation-cluster`, we direct the request to federation apiserver.
+we can ensure that the cluster registration was successful by running:
+
+```shell
+$ kubectl get clusters --context=federation-cluster
+NAME       STATUS    VERSION   AGE
+cluster1   Ready               3m
+```
+
+### Updating KubeDNS
+
+Once the cluster is registered with federation, we are all ready to use it.
```

Comment:
You could also say "with the federation" if you like that better. Just not "with federation" which is not grammatically correct. :)